.github/workflows/compliance-ncs.yml

```diff
@@ -0,0 +1,63 @@
+name: Compliance NCS
+
+on: pull_request
+
+jobs:
+  compliance_job:
+    runs-on: ubuntu-latest
+    name: Run compliance checks on patch series (PR)
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v2
+        with:
+          path: ncs/zephyr
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+
+      - name: cache-pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-doc-pip
+
+      - name: Run Compliance Tests
+        env:
+          BASE_REF: ${{ github.base_ref }}
+        working-directory: ncs
+        run: |
+          pip3 install -U pip
+          pip3 install -U setuptools
+          export PATH="$HOME/.local/bin:$PATH"
+          pip3 install -U west
+          west init -m https://github.com/nrfconnect/sdk-nrf
+          west update zephyr nrfxlib
+          pip3 install -U gitlint
+          pip3 install -r zephyr/scripts/requirements-compliance.txt
+          export ZEPHYR_BASE=$(pwd)/zephyr
+          cd zephyr
+          ./scripts/ci/check_compliance.py -m Codeowners -m Gitlint -m Identity -m Nits -m pylint -m checkpatch -m Kconfig -c origin/${BASE_REF}.. || true
```

Comment:
You will need to remove `-m Kconfig` here. If you try locally you will see it fails because of missing Kconfig options that are defined in sdk-nrf

Reply:
Hmmm, I tried locally and had no issue. The Kconfig compliance test sets `KCONFIG_BINARY_DIR` to `/tmp` and writes `Kconfig.modules` with the following contents:

```
osource "/home/utzig/src/Nordic/ncs/nrf/Kconfig.nrf"
osource "/home/utzig/src/Nordic/ncs/bootloader/mcuboot/zephyr/Kconfig"
osource "/home/utzig/src/Nordic/ncs/nrfxlib/Kconfig.nrfxlib"
```

I assumed this is how all external modules were "glued" together. What error did you see running locally?

Reply:
Well, I can see how `sdk-nrf` and `sdk-zephyr` could get out of sync in their config options. Given that `west.yml` from `sdk-nrf` is used, it might have options that would only match some other `sdk-zephyr` branch, but I think this would also affect `sdk-nrfxlib`. The only reliable place to test `-m Kconfig` is `sdk-nrf` since it needs to bring in the appropriate modules. Also the whole point of using `west update -k` was for running `-m Kconfig`, since every other compliance check is kinda self-contained, so only Zephyr itself is needed because the scripts are there. So I could as well just grab `sdk-zephyr`, what do you think?